src/sync/events/utils.ts

```diff
@@ -26,15 +27,21 @@ export const fetchBlock = async (chainId: number, blockNumber: number, force = f
         } else {
           const nativeTokenTransaction: es.ftTransfers.Event[] = [];
           const block = await getProvider(chainId).getBlockWithTransactions(blockNumber);
+
           // Create transactions array to store
           const transactions = (block.transactions as ITransactionResponse[]).map(
             (tx: ITransactionResponse) => {
               // eslint-disable-next-line @typescript-eslint/no-explicit-any
               const rawTx = tx.raw as any;
 
               const gasPrice = tx.gasPrice?.toString();
-              const gasUsed = rawTx?.gas ? bn(rawTx.gas).toString() : undefined;
+              const gasLimit = tx.gasLimit?.toString();
+
+              let gasUsed = rawTx?.gas ? bn(rawTx.gas).toString() : undefined;
+              if (!gasUsed) gasUsed = gasLimit;
```

Comment:
Lets do not do this. GasUsed and gas limit both are different things. Instead, you can add gasLimit column and show it as it is.